include/nbl/asset/IRenderpassIndependentPipeline.h

```diff
@@ -111,6 +124,18 @@ struct SVertexInputParams
     //! index in array denotes binding number
 	SVertexInputBindingParams bindings[MAX_ATTR_BUF_BINDING_COUNT] = {};
 
+    std::string to_string() const
+    {
+        std::string cacheKey;
+
+        cacheKey += ICacheKeyCreator::getHexString(enabledAttribFlags) + ICacheKeyCreator::getHexString(enabledBindingFlags);
+        for (uint32_t i = 0; i < MAX_VERTEX_ATTRIB_COUNT; ++i)
+            cacheKey += attributes[i].to_string();
+        for (uint32_t i = 0; i < MAX_ATTR_BUF_BINDING_COUNT; ++i)
+            cacheKey += bindings[i].to_string();
```

Comment:
this might bite you in the ass here
https://github.com/Devsh-Graphics-Programming/Nabla/pull/9/files?file-filters%5B%5D=.cpp&file-filters%5B%5D=.glsl&file-filters%5B%5D=.h&file-filters%5B%5D=No+extension#r608493309